design/00292: EEPROM storage.md

```diff
@@ -0,0 +1,64 @@
+# Proposal: Use the PlanktoScope HAT EEPROM chip to store hardware configuration
+
+Author(s): Vallet, Laurent; Le Corre, Cléa Last updated: 30/10/2024 Discussion at https://github.com/PlanktoScope/PlanktoScope/issues/292
+
+## Abstract
+
+The following document proposes to store the PlanktoScope (PS) hardware configuration in the 32 Kbits EEPROM chip of the PlanktoScope HAT in order to : 
+* automatically save the hardware state of the machine
+* easily retrieve the hardware configuration for debugging
+* feed the metadata with precise hardware information
+
+## Background
+
+A specific HAT was developped to satisfy the PlanktoScope needs. 
+This PlanktoScope HAT connected to the raspberry pi and all of the external devices (led, motors, pump, camera) contains a 32 Kbit EEPROM chip which is a "type of non-volatile ROM that enables individual bytes of data to be erased and reprogrammed" (Wikipedia).
+It is therefore possible to store some data inside this chip and to read it back.
+
+For now (Sunray version 2024.0.0-beta.2) some of the hardware configuration of the PlanktoScope is entered by the user in the GUI.
+We would like to have an hardware configuration specific to each machine in case some specific modification is made.
+We would like to store this configuration in a safe place, independent from the software installed on the raspberry pi.
+
+## Proposal
+
+We therefore propose to write a python script that allows the storage and the reading of the PlanktoScope hardware information on the EEPROM chip.
+
+This information stored should contain : 
+* PlanktoScope reference
+* Plantoscope serial number
+* PlanktoScope version
+* Plantoscope fabrication date
+* HAT serial number
+* HAT version
+* Driver reference
+* Pump reference
+* Stepper focus reference
+* Objective lens reference
+* Tube lens reference
+* Flowcell thickness
+* LED reference
+
+Those informations could be used to feed the metadata stored in the hardware.json file as described in the issue 290 (https://github.com/PlanktoScope/PlanktoScope/issues/290).
+
```

Comment:
Since you are adding MQTT API routes, please also describe the design of the MQTT API routes you are adding.